framework/include/materials/MaterialPropertyStorage.h

```diff
@@ -285,6 +285,9 @@ class MaterialPropertyStorage
                  const Elem & elem,
                  unsigned int side,
                  unsigned int n_qpoints);
+
+  // Need to be able to eraseProperty from here
+  friend class ProjectMaterialProperties;
```

Comment:
If possible see if you can narrow the friendship down to the specific method you need to call. That way we keep these friend declarations to the minimum. You can't always do this because it requires the full declaration of the other class.